File: AIproyect/AI.py
class TreeNode:
    def __init__(self, peso=1, palabraroot=""):
        self.peso = peso
        self.palabraroot = palabraroot
        self.left = None
        self.right = None
        self.opcion = None

    def init(self, pesoParam, palabrarootParam):
        self.peso = pesoParam
        self.palabraroot = palabrarootParam

        if self.opcion is None:
            self.opcion = TreeNode(pesoParam, palabrarootParam)
        elif self.opcion.peso <= pesoParam and self.right.opcion is None:
            self.right.opcion = TreeNode(pesoParam, palabrarootParam)
        elif self.opcion.peso > pesoParam:
            if self.left.opcion is None:
                self.left.opcion = TreeNode(pesoParam, palabrarootParam)
            else:
                self.left.left.opcion = TreeNode(pesoParam, palabrarootParam)
        elif self.right.opcion.peso < pesoParam:
            if self.right.right.opcion is None:
                self.right.right.opcion = TreeNode(pesoParam, palabrarootParam)
            else:
                self.right.right.right.opcion = TreeNode(pesoParam, palabrarootParam)
        else:
            if self.right.left.opcion is None:
                self.right.left.opcion = TreeNode(pesoParam, palabrarootParam)
            else:
                self.right.left.left.opcion = TreeNode(pesoParam, palabrarootParam)

    def LTM(self, palabra):
        if self.palabraroot == palabra:
            return self
        elif self.left and self.left.palabraroot <= palabra:
            return self.left.LTM(palabra)
        elif self.right and self.right.palabraroot > palabra:
            return self.right.LTM(palabra)
        else:
            return None


if __name__ == "__main__":
    print("Procesando")

    Cerebro = None
    peso = 1

    with open("0_palabras_todas.txt", "r", errors="ignore") as file:
        for line in file:
            palabra = line.strip()  # Elimina los espacios en blanco al principio y al final

            if Cerebro is None:
                Cerebro = TreeNode(peso, palabra)
            else:
                Cerebro.init(peso, palabra)
                peso += 1
    palabra_buscar = input("Por favor, introduce una palabra para buscar en Cerebro: ")
    nodo = Cerebro.LTM(palabra_buscar)
    if nodo:
        print(f"La palabra '{palabra_buscar}' fue encontrada en Cerebro con un peso de {nodo.peso}.")
    else:
        print(f"La palabra '{palabra_buscar}' no fue encontrada en Cerebro.")
    print("Proceso terminado, se procesaron todas las palabras del archivo")

File: AIproyect/AI.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>

using namespace std;

class TreeNode {
public:

    int peso = 1;
    string palabraroot = "";
    TreeNode *left = new TreeNode();
    TreeNode *right = new TreeNode();
    TreeNode *opcion = new TreeNode();

    TreeNode(){

    }

void setup(int pesoParam, string palabrarootParam) : peso(pesoParam), palabraroot(palabrarootParam) {
    if(left->palabraroot == ""){
        left->palabraroot = palabrarootParam;
    }else if(left->palabraroot <= palabrarootParam && right->palabraroot == ""){
        right->palabraroot = palabraroot;
    }else if(left->palabraroot > palabrarootParam){
        if(left->left == nullptr) {
            left->left = new TreeNode();
        }
        left->left->palabraroot = palabrarootParam;
    }else if(right->palabraroot < palabrarootParam){
        if(right->right == nullptr) {
            right->right = new TreeNode();
        }
        right->right->palabraroot = palabrarootParam;
    }else{
        if(right->left == nullptr) {
            right->left = new TreeNode();
        }
        right->left->palabraroot = palabrarootParam;
    }
}

void Init(int pesoParam, string palabrarootParam) {

    if(opcion == nullptr){
        opcion->peso = pesoParam;
        opcion->palabraroot = palabrarootParam;
        right->left->opcion->palabraroot = palabrarootParam;
    }else if(opcion->peso <= pesoParam && right->opcion == nullptr){
        right->opcion->palabraroot = palabrarootParam;
    }else if(opcion->peso > pesoParam){
        if(left->opcion == nullptr){
            left->opcion->palabraroot = palabrarootParam;
            right->left->opcion->peso = pesoParam;
        }else{
            left->left->opcion->palabraroot = palabrarootParam;
            right->left->opcion->peso = pesoParam;
        }
    }else if(right->opcion->peso < pesoParam){
        if(right->right->opcion == nullptr){
            right->right->opcion->peso = pesoParam;
            right->left->opcion->peso = pesoParam;
        }else{
            right->right->right->opcion->palabraroot = palabrarootParam;
            right->left->opcion->peso = pesoParam;
        }
    }else{
        if(right->left->opcion == nullptr){
            right->left->opcion->palabraroot = palabrarootParam;
            right->left->opcion->peso = pesoParam;
        }else{
            right->left->left->opcion->palabraroot = palabrarootParam;
            right->left->left->opcion->peso = pesoParam;
        }
    }
}




void LTM(){
  
}

};

/*
class Cerebro {
public:
    TreeNode arbol1;
    void setup(string palabra) {
        // Aquí puedes agregar el código para configurar el cerebro
    }
};
*/
int main() {

    cout << "procesando" << endl;

    ifstream file("0_palabras_todas.txt");
    if (!file.is_open()) {
        cout << "No se pudo abrir el archivo palabras.txt\n";
        return 1;
    }

    TreeNode* Cerebro = nullptr;
    string line;
    int peso = 1;

    string palabra;
    for (int i = 0; i < 10 && getline(file, line); i++) { 
        if (Cerebro == nullptr) {
                Cerebro->setup(peso, line);
                //Cerebro->Init(peso, line);
            } else {
                // Aquí va el código para agregar la palabra al árbol Cerebro
            }
            peso++;
    }
    cout << "proceso terminado, se leyeron todas las palabras" << endl;
    return 0;
}
